api/secrets/secret_handler.go

```diff
@@ -0,0 +1,616 @@
+	componentSecretsMap := make(map[string]map[string]bool)
+	for _, component := range activeDeployment.Spec.Components {
+		secrets := eh.getSecretsForComponent(&component)
+		if len(secrets) <= 0 {
+			continue
+		}
+		componentSecretsMap[component.Name] = secrets
+	}
+	for _, job := range activeDeployment.Spec.Jobs {
+		secrets := eh.getSecretsForComponent(&job)
+		if len(secrets) <= 0 {
+			continue
+		}
+		componentSecretsMap[job.Name] = secrets
+	}
+
+	secretDTOsMap := make(map[string]models.Secret)
+	for componentName, secretNamesMap := range componentSecretsMap {
+		secretObjectName := k8sObjectUtils.GetComponentSecretName(componentName)
+
+		secret, err := eh.client.CoreV1().Secrets(envNamespace).Get(context.TODO(), secretObjectName, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			// Mark secrets as Pending (exist in config, does not exist in cluster) due to no secret object in the cluster
+			for secretName := range secretNamesMap {
+				secretNameAndComponentName := fmt.Sprintf("%s-%s", secretName, componentName)
+				if _, exists := secretDTOsMap[secretNameAndComponentName]; !exists {
+					secretDTO := models.Secret{Name: secretName, DisplayName: secretName, Component: componentName, Status: models.Pending.String(), Type: models.SecretTypeGeneric}
+					secretDTOsMap[secretNameAndComponentName] = secretDTO
+				}
+			}
+			continue
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		// Secrets from cluster
+		clusterSecretEntriesMap := secret.Data
+
+		// Handle Pending secrets (exist in config, does not exist in cluster) due to no secret object entry in the cluster
+		for secretName := range secretNamesMap {
+			secretNameAndComponentName := fmt.Sprintf("%s-%s", secretName, componentName)
+			if _, exists := secretDTOsMap[secretNameAndComponentName]; exists {
+				continue
+			}
+			status := models.Consistent.String()
+			if _, exists := clusterSecretEntriesMap[secretName]; !exists {
+				status = models.Pending.String()
+			}
+			secretDTO := models.Secret{Name: secretName, DisplayName: secretName, Component: componentName, Status: status, Type: models.SecretTypePending}
+			secretDTOsMap[secretNameAndComponentName] = secretDTO
+		}
+
+		// Handle Orphan secrets (exist in cluster, does not exist in config)
+		for clusterSecretName := range clusterSecretEntriesMap {
+			clusterSecretNameAndComponentName := fmt.Sprintf("%s-%s", clusterSecretName, componentName)
+			if _, exists := secretDTOsMap[clusterSecretNameAndComponentName]; exists {
+				continue
+			}
+			status := models.Consistent.String()
+			if _, exists := secretNamesMap[clusterSecretName]; !exists {
+				status = models.Orphan.String()
+			}
+			secretDTO := models.Secret{Name: clusterSecretName, DisplayName: clusterSecretName, Component: componentName, Status: status, Type: models.SecretTypeOrphaned}
```

Comment:
Why is Type set to SecretTypeOrphaned and not SecretTypeGeneric?

Reply:
Explained in [here](https://github.com/equinor/radix-api/pull/374#discussion_r786642021)